scripts/rv_expression_association/get_rv_outliers.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+
+import click
+import logging
+import hail as hl
+import pandas as pd
+from cpg_utils.hail_batch import dataset_path, init_batch, output_path, reference_path
+from cloudpathlib import AnyPath
+
+# use logging to print statements, display at info level
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+MT = dataset_path('mt/v7.mt')
+
+
+# get OneK1K sample ID and gene name as arguments using click
+@click.command()
+@click.option('--onek1k-id', required=True)
+@click.option('--gene-name', required=True)
+@click.option('--sample-mapping-file', required=True)
+@click.option('--gene-file', required=True)
+@click.option(
+    '--window-size',
+    required=True,
+    help='the size of the flanking region, to be added twice, on either side of the gene body',
+)
+def main(
+    onek1k_id: str,
+    gene_name: str,
+    sample_mapping_file: str,  # 'scrna-seq/grch38_association_files/OneK1K_CPG_IDs.tsv'
+    gene_file: str,  # e.g., 'scrna-seq/grch38_association_files/gene_location_files/GRCh38_geneloc_chr1.tsv'
+    window_size: int,
+):
+    """for a given individual,
+    and gene for which that individual is an expression outlier
+    get all variants within a given window,
+
+    then filter for variants that:
+    - are alt (0/1 or 1/1) for that individual
+    - are biallelic SNVs
+    - have regulatory consequences based on VEP annotations
+
+    also annotate variants with the following:
+    - CADD score
+    - MAF within the OneK1K sample
+    - MAF in gnomad
+
+    create the following table:
+    donor ID | gene ID | variant ID | position | CADD | MAF (OneK1K) | MAF (gnomad) | ..
+    """
+
+    # file matching OneK1K IDs to CPG (internal) and TOB (external) IDs
+    sample_key_df = pd.read_csv(
+        AnyPath(dataset_path(sample_mapping_file)), sep='\t', index_col=0
+    )
+
+    cpg_id = sample_key_df[sample_key_df.index == onek1k_id]['InternalID'].values[0]
+    logging.info(f'CPG ID: {cpg_id}')  # e.g., 'CPG9951'
+
+    # define output filename and check if it already exists
+    output_filename = AnyPath(output_path(f'{cpg_id}_{gene_name}_{window_size}.csv'))
+    logging.info(f'Output file: {output_filename}')
+
+    # skip if file already exists
+    if output_filename.exists():
+        raise Exception(f'File {output_filename} already exists, exiting')
+
+    init_batch()
+    # get VEP-annotated WGS object (hail matrix table)
+    # mt = hl.read_matrix_table('gs://cpg-tob-wgs-test/tob_wgs_vep/v1/vep105_GRCh38.mt')
+    mt = hl.read_matrix_table(MT)
```

Comment:
At least the currently set `MT` doesn't have VEP annotations.

Reply:
Yes, certainly, that's probably something you should coordinate with @KatalinaBobowik. There's also a discussion about the GENCODE / VEP version here: https://centrepopgen.slack.com/archives/C018KFBCR1C/p1661468741596349?thread_ts=1658208231.938539&cid=C018KFBCR1C